src/content-tags-drawer/data/apiHooks.jsx

```diff
@@ -109,3 +114,13 @@ export const useContentDataResponse = (contentId) => {
 export const useIsContentDataLoaded = (contentId) => (
   useContentData(contentId).status === 'success'
 );
+
+/**
+ * Builds the mutation to update the tags applied to the content object
+ * @returns {import("./types.mjs").UseMutationResult}
```

Comment:
Can we delete this line `@returns {import("./types.mjs").UseMutationResult}` and the custom `UseMutationResult` type? TypeScript can infer the correct type from the react-query code, and the full type is much more useful/accurate.

With the current code:
![before](https://github.com/openedx/frontend-app-course-authoring/assets/945577/910f320e-872f-4c23-bfe9-39fcc138c74e)

Without that line you actually get the full, accurate type:
![Screenshot 2023-11-23 at 10 33 31 AM](https://github.com/openedx/frontend-app-course-authoring/assets/945577/58f2df65-acd5-42b0-b22f-8213c702ca35)


## No need for useContentTaxonomyTagsDataResponse

Also, there is a similar issue above. The return type of `useContentTaxonomyTagsData` is wrong, and should be set to `@returns {import("@tanstack/react-query").UseQueryResult<import("./types.mjs").ContentTaxonomyTagsData>}`.

If you do that, you'll see there's no need for the `useContentTaxonomyTagsDataResponse` function at all. 

Instead of

```javascript
const useContentTaxonomyTagsData = () => {
    const contentTaxonomyTagsData = useContentTaxonomyTagsDataResponse(contentId);
    const isContentTaxonomyTagsLoaded = useIsContentTaxonomyTagsDataLoaded(contentId);
    return { contentTaxonomyTagsData, isContentTaxonomyTagsLoaded };
  };
  const { contentTaxonomyTagsData, isContentTaxonomyTagsLoaded } = useContentTaxonomyTagsData();
```

With the accurate types in place, you can just write:


```javascript
const {data, isSuccess} = useContentTaxonomyTagsData(...);
```

Or if you want to rename those variables, write it as:
```javascript
const {data: contentTaxonomyTagsData, isSuccess: isContentTaxonomyTagsLoaded} = useContentTaxonomyTagsData(...);
```
which is exactly the same thing as the code you have now, but with way less boilerplate.

![screenshot](https://github.com/openedx/frontend-app-course-authoring/assets/945577/2acc8a70-169b-4824-bc2d-134ecc35139f)

Reply:
Interesting! Thanks for the detailed breakdown. That's definitely much simpler and cleaner, I updated all of them to follow this structure: https://github.com/openedx/frontend-app-course-authoring/pull/704/commits/bdc8d3026150400e6d7b081e71e61180b83d5ff3

Reply:
Nice, much simpler :)